jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-backend/src/main/java/org/jbpm/workbench/cm/backend/server/RemoteCaseManagementServiceImpl.java

```diff
@@ -159,5 +172,81 @@ public void removeComment(final String serverTemplateId, final String containerI
         return comparing(CaseMilestoneSummary::getStatus).thenComparing(request.getSortByAsc() ? comparatorByName: comparatorByName.reversed());
     }
 
+    @Override
+    public Actions getCaseActions(String serverTemplateId, String container, String caseId, String userId) {
+        Actions actions = new Actions();
+        actions.setAvailableActions(getAdHocActions(serverTemplateId, container, caseId));
+        actions.setInProgressAction(getInProgressActions(container, caseId));
+        actions.setCompleteActions(getCompletedActions(container, caseId));
+        return actions;
+    }
+
+    public List<CaseActionSummary> getInProgressActions(String containerId, String caseId) {
+        List<NodeInstance> activeNodes = client.getActiveNodes(containerId, caseId, 0, PAGE_SIZE_UNLIMITED);
+
+        return activeNodes.stream()
```

Comment:
as we do a service call to kie-server, perhaps we could use `parallelStream()` in here.